guardian/amiable
# Add filtering by account name.

## What does this change?
Adds a dropdown to amiable allowing you to filter instances by account name and updates the AMI display to also show account name. 

The motivation here is that AMIable needs to become a source of truth for all AMIs in use at the guardian - even if they aren't properly tagged - and account name is a piece of information we can reliably have about each instance. Whilst there's been some excellent work done on https://github.com/guardian/tag-janitor we can't guarantee that stuff will be properly tagged, so I'm trying to adjust the UI to allow for this. 

The extra 'account' column causes the boxes to get a bit wide unfortunately, so I've had to give them a max width and allow word breaking in the CSS for the table. I've tried to limit the impact of this by only showing the account column where accountName != stackName

 It looks like this at the moment:
<img width="1208" alt="Screenshot 2020-12-01 at 18 16 53" src="https://user-images.githubusercontent.com/3606555/100782290-1dd5da80-3404-11eb-98a3-73e479a5729b.png">


<img width="568" alt="Screenshot 2020-12-01 at 18 27 24" src="https://user-images.githubusercontent.com/3606555/100781414-e7e42680-3402-11eb-98b7-5f3302523814.png">


Next steps:
 - Use the /stack, /stage, /app endpoints in prism to make those fields dropdowns or autocomplete fields
 - Investigate exposing package level information about AMIs - e.g. "what version of nginx is this AMI running" 

## How to test

Run amiable, check that you can see the account filter, that it works, and that it doesn't break the UI.

## How can we measure success?
Long term, number of out of date AMIs should go down. A proper measure would be to see how many times the account filter gets used, but I think I'll just ask people if they've used it...


## Have we considered potential risks?


## Images
